operator/internal/controller/pulumi/stack_controller_test.go

```diff
@@ -623,12 +711,27 @@ var _ = Describe("Stack Controller", func() {
 					LastResyncTime:       metav1.Now(),
 					LastAttemptedCommit:  fluxRepo.Status.Artifact.Digest,
 					LastSuccessfulCommit: "",
+					Failures:             2,
 				}
 			})
-			It("reconciles", func(ctx context.Context) {
-				_, err := reconcileF(ctx)
-				Expect(err).NotTo(HaveOccurred())
-				ByResyncing()
+			When("within cooldown period", func() {
+				It("backs off exponentially", func(ctx context.Context) {
+					res, err := reconcileF(ctx)
+					Expect(err).NotTo(HaveOccurred())
+					// 5 minutes * 2^2
+					Expect(res.RequeueAfter).To(BeNumerically("~", time.Duration(20*time.Minute), time.Minute))
+					ByMarkingAsReady()
```

Comment:
Why would the stack be ready when it is in backoff?  Consider, for example, the Stack dependencies feature, where one stack waits for another to be ready.  You wouldn't want to unblock a dependent.

Reply:
Changed to reconciling.